src/main/java/com/google/devtools/common/options/Converters.java

```diff
@@ -646,17 +646,21 @@ public String getTypeDescription() {
    * A converter for variable assignments from the parameter list of a blaze command invocation.
    * Assignments are expected to have the form "name[=value]", where names and values are defined to
    * be as permissive as possible and value part can be optional (in which case it is considered to
-   * be null).
+   * be null). The special syntax "=name" is also supported and recorded as a null key with value
+   * set to the name. Consumers should interpret this as a request to remove any previously assigned
+   * value for that name.
    */
   public static class OptionalAssignmentConverter
       extends Converter.Contextless<Map.Entry<String, String>> {
```

Comment:
WDYT about using a richer payload type here, say, something like

```
public sealed interface EnvVar {
  record Set(String key, String value) implements EnvVar {}
  record Inherit(String key) implements EnvVar {}
  record Unset(String key) implements EnvVar {}
}
```

so that it's more difficult for a caller to misinterpret the meaning of a `null` key or value?